react-native-cli/__tests__/cli-test.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+jest.mock('fs');
+jest.mock('child_process');
+jest.dontMock('../index.js');
+jest.setMock('prompt', {
+  start: jest.genMockFn(),
+  get: jest.genMockFn()
+});
+
+describe('react-native cli', function() {
+  var originalArgs;
+  var originalLog;
+  var originalError;
+  var originalExit;
+  var originalChdir;
+
+  beforeEach(function() {
+    originalArgs = process.argv;
+    originalLog = console.log;
+    originalError = console.error;
+    originalExit = process.exit;
+    originalChdir = process.chdir;
+
+    console.log = jest.genMockFn();
+    console.error = jest.genMockFn();
+    process.exit = jest.genMockFn();
```

Comment:
fyi @martinbigio works on new CLI toolkit where each command is a function that accepts `argv` array and returns a promise, which makes it much easier to test – no need for all these mocks.

Not sure if you want to wait for that to land though, should be fairly easy to convert your tests to new system.